sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -1229,6 +1229,22 @@ case class Window(
     copy(child = newChild)
 }
 
+case class WindowGroupLimit(
+    partitionSpec: Seq[Expression],
+    orderSpec: Seq[SortOrder],
+    rankLikeFunction: Expression,
+    limit: Int,
+    child: LogicalPlan) extends OrderPreservingUnaryNode {
```

Comment:
why it can be an `OrderPreservingUnaryNode`, should the ordering be affected by `orderSpec` ?

Reply:
Good point. `WindowGroupLimit` will sort the input data. @beliefer please fix this.